### PR TITLE
add --with-swig=<path to swig> option to configure

### PR DESCRIPTION
**Pull request by spfennell**

### CODE REVIEW

**Comment by spfennell** [comment]
@dbankieris I hear you have a test case

File: configure [hunk 1]
```diff
@@ -1385,6 +1387,7 @@ Optional Features:
 Optional Packages:
   --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
   --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
+  --with-swig[=DIR]       SWIG root directory
```
Comment (dbankieris) [inline configure]: One of my frustrations with `configure` has been the vagueness of option descriptions. I would prefer something less ambiguous here, such as `directory containing the swig executable`.

Comment (spfennell) [reply configure]: I agree. I was just trying to conform! Maybe I'll update some of the other help strings.

Comment (dbankieris) [reply configure]: That would be great!

**Pull request merged**